apisix/plugins/real-ip.lua

```diff
@@ -67,6 +73,19 @@ function _M.check_schema(conf)
 end
 
 
+local function addr_match(conf, addr)
+    if not conf.trusted_addresses then
+        return false
+    end
+
+    if not conf.matcher then
+        conf.matcher = core.ip.create_ip_matcher(conf.trusted_addresses)
+    end
```

Comment:
It is bad practice to add properties to conf, and we should keep conf pure. How about making `matcher` a module-level variable?

Reply:
Changed it to local variable.